cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/SimpleKafkaProducer.java

```diff
@@ -53,19 +58,44 @@ public SimpleKafkaProducer(CConfiguration cConf) {
 
     ProducerConfig config = new ProducerConfig(props);
     kafkaTopic = cConf.get(Constants.Logging.KAFKA_TOPIC);
-    producer = new Producer<>(config);
+    producer = createProducer(config);
   }
 
   public void publish(String key, byte[] bytes) {
+    // Clear the interrupt flag, otherwise it won't be able to publish
+    boolean threadInterrupted = Thread.interrupted();
     try {
       KeyedMessage<String, byte[]> data = new KeyedMessage<>(kafkaTopic, key, bytes);
       producer.send(data);
     } catch (Throwable t) {
       LOG.error("Exception when trying to publish log message to kafka with key {} and topic {}", key, kafkaTopic, t);
+    } finally {
+      // Reset the interrupt flag if needed
+      if (threadInterrupted) {
+        Thread.currentThread().interrupt();
```

Comment:
Hum? It already has the comment "Reset the interrupt flag if needed". It is based on the `threadInterrupted` local variable that get set right above.